esmf_regrid/tests/unit/experimental/unstructured_scheme/test_MeshToGridESMFRegridder.py

```diff
@@ -397,3 +397,79 @@ def test_curvilinear():
     # Check metadata and scalar coords.
     result.data = expected_data
     assert expected_cube == result
+
+
+def test_masks():
+    """
+    Test initialisation of :func:`esmf_regrid.experimental.unstructured_scheme.MeshToGridESMFRegridder`.
+
+    Checks that the `src_mask` and `tgt_mask` keywords work properly.
+    """
+    src = _gridlike_mesh_cube(7, 6)
+    tgt = _curvilinear_cube(6, 7, [-180, 180], [-90, 90])
+
+    src_mask = np.zeros([6 * 7], dtype=bool)
+    src_mask[0] = True
+    src.data = np.ma.array(src.data, mask=src_mask)
+
+    # Make tgt discontiguous at (0, 0)
+    tgt_mask = np.zeros([7, 6], dtype=bool)
+    tgt_mask[0, 0] = True
+    tgt.data = np.ma.array(tgt.data, mask=tgt_mask)
+    tgt_discontiguous = tgt.copy()
+    tgt_discontiguous.coord("latitude").bounds[0, 0] = 0
+    tgt_discontiguous.coord("longitude").bounds[0, 0] = 0
+
+    rg_src_masked = MeshToGridESMFRegridder(src, tgt, src_mask=True)
+    rg_tgt_masked = MeshToGridESMFRegridder(src, tgt_discontiguous, tgt_mask=True)
+    rg_unmasked = MeshToGridESMFRegridder(src, tgt)
+
+    weights_src_masked = rg_src_masked.regridder.weight_matrix
+    weights_tgt_masked = rg_tgt_masked.regridder.weight_matrix
+    weights_unmasked = rg_unmasked.regridder.weight_matrix
+
+    # Check there are no weights associated with the masked point.
+    assert weights_src_masked[:, 0].nnz == 0
+    assert weights_tgt_masked[0].nnz == 0
+
+    # Check all other weights are correct.
+    assert np.allclose(
+        weights_src_masked[:, 1:].todense(), weights_unmasked[:, 1:].todense()
+    )
+    assert np.allclose(weights_tgt_masked[1:].todense(), weights_unmasked[1:].todense())
+
+
+def test_masks_with_resolution():
```

Comment:
This is a prime case for [PyTest parametrization](https://docs.pytest.org/en/6.2.x/parametrize.html) - for ease of interpretation and maintenance.